.gitlab-ci.yml

```diff
@@ -0,0 +1,268 @@
+    key: $MARIADB_MAJOR_VERSION-$CI_JOB_NAME
+    paths:
+      - .ccache
+    # policy: pull
+    # @TODO: It would be enough to only download the cache. There is no need for
+    # every job to upload a new cache every time. A monthly or weekly scheduled
+    # run could update the cache for all other builds to us.
+
+centos8:
+  stage: build
+  image: centos:8
+  variables:
+    GIT_STRATEGY: fetch
+    GIT_SUBMODULE_STRATEGY: normal
+  script:
+    - yum install -y yum-utils rpm-build openssl-devel
+    - yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
+    # dnf --enablerepo=powertools install Judy-devel  #--> not found
+    - dnf config-manager --set-enabled powertools
+    # Error:
+    # Problem: conflicting requests
+    # - package Judy-devel-1.0.5-18.module_el8.3.0+757+d382997d.i686 is filtered out by modular filtering
+    # - package Judy-devel-1.0.5-18.module_el8.3.0+757+d382997d.x86_64 is filtered out by modular filtering
+    # Solution: install Judy-devel directly from downloaded rpm file:
+    - yum install -y http://mirror.centos.org/centos/8/PowerTools/x86_64/os/Packages/Judy-devel-1.0.5-18.module_el8.3.0+757+d382997d.x86_64.rpm
+    # Use eatmydata to speed up build
+    - yum install -y https://github.com/stewartsmith/libeatmydata/releases/download/v129/libeatmydata-129-1.fc33.x86_64.rpm
+    - yum install -y ccache  # From EPEL
+    - source /etc/profile.d/ccache.sh
+    - export CCACHE_DIR="$(pwd)/.ccache"; ccache -s
+    - yum-builddep -y mariadb-server
+    - mkdir builddir; cd builddir
+    - cmake -DRPM=$CI_JOB_NAME $CMAKE_FLAGS .. 2>&1 | tee -a ../build-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+    - eatmydata make package 2>&1 | tee -a ../build-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+    # @TODO: Don't use -j on Gitlab.com as builds just get stuck when running
+    # multi-proc, needs more debugging
+    - make test || true  # Unit tests constantly fail, see https://jira.mariadb.org/browse/MDEV-25820
+    # - make test-force || true  # mysql-test-runner takes too long, run it in a separate job instead
+    - *rpm_listfiles
+    - mkdir ../rpm; mv *.rpm ../rpm
+    - ccache -s
+  artifacts:
+    when: always  # Must be able to see logs
+    paths:
+      - build-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+      - rpmlist-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+      - rpm
+      - builddir/_CPack_Packages/Linux/RPM/SPECS/
+  cache:
+    key: $MARIADB_MAJOR_VERSION-$CI_JOB_NAME
+    paths:
+      - .ccache
+
+centos7:
+  stage: build
+  image: centos:7
+  variables:
+    GIT_STRATEGY: fetch
+    GIT_SUBMODULE_STRATEGY: normal
+  script:
+    - yum install -y yum-utils rpm-build gcc gcc-c++ bison libxml2-devel libevent-devel openssl-devel
+    - yum-builddep -y mariadb-server
+    - mkdir builddir; cd builddir
+    - cmake -DRPM=$CI_JOB_NAME $CMAKE_FLAGS .. 2>&1 | tee -a ../build-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+    - make package 2>&1 | tee -a ../build-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
```

Comment:
Instead of this I would use `cmake --build . --target package` (note `cmake` version I think 3.19+ should be used) since it is portable and goes along with other generators (you can use it in other systems).

Reply:
There are many ways to run it. Why do you suggest `cmake --build . --target package` would be the best way, what are the advantages or differences to `make package`?

- page https://mariadb.com/kb/en/source-building-mariadb-on-centos/ says `make package`
- page https://mariadb.com/kb/en/building-mariadb-on-fedora/ says also `make package`
- page https://mariadb.com/kb/en/generic-build-instructions/ says `make package`
- page https://mariadb.org/get-involved/getting-started-for-developers/get-code-build-test/ says just `cmake --build` without proceeding to the `install` or `package` steps
- page https://mariadb.com/kb/en/get-build-and-test-latest-mariadb-the-lazy-way/ says `make install` without proceeding to the `package` stage

I would argue that the .gitlab-ci.yml needs to do the "familiar" steps. It will be confusing to a random new contributor if all docs say to do it in one way, but then the CI itself actually would not follow the docs but do it in another way.

Reply:
Well I started using recently `cmake --build .` because I had to some patches in Windows what is the suggested approach (look https://mariadb.com/kb/en/Building_MariaDB_on_Windows/).
Short answer: it is portable way ->  `cmake`'s way and I'm suggesting to use it and to change documentation. 
As a generator I'm using `ninja` so `make package` wouldn't work. 
In that case `ninja package` would, but in order not to be dependent of type of used generator, -> use `cmake` 's way: `cmake --build . --target package`.
Documentation : `cmake --build . --help`. 
Default build targets are dependent of the type of [generated build system](https://stackoverflow.com/questions/30532448/what-is-the-name-of-cmakes-default-build-target)

Reply:
In this case we don't use Ninja but generate GNU Makefiles. A third way would be to run cpack directly. These are however very marginal nuances and does not affect really the decision on if this .gitlab-ci.yml is correctly designed and implemented.